File: src/Java/SelfNumber.java
package Java;

/**
 * jhChoi - 201119
 * 셀프넘버
 */
public class SelfNumber {
    public static void main(String[] args) {
        boolean[] selfNum = new boolean[10001];

        for (int i = 1; i <= 10000; i++) {
            if (selfNum[i]) {
                continue;
            }

            int num = i;
            while (num < 10001) {
                char[] arr = Integer.toString(num).toCharArray();
                for (char c : arr) {
                    num = num + (c - '0');
                }

                if(num >10000){
                    break;
                }
                selfNum[num] = true;
            }
        }

        for (int i = 1; i < selfNum.length; i++) {
            if(!selfNum[i]) System.out.println(i);
        }
    }
}

File: src/Kotlin/SelfNumber.kt
package Kotlin

fun main() {
    val selfNum: BooleanArray = BooleanArray(10001)

    for (i in 1..10000) {
        if (selfNum[i]) continue

        var num: Int = i;
        while (num <= 10000) {
            val arr: CharArray = Integer.toString(num).toCharArray()
            for (n in arr) num += (n - '0')
            if (num > 10000) break;
            selfNum[num] = true
        }
    }

    for (i in 1 until selfNum.size){
        if(!selfNum[i]) println(i)
    }
}